apps/election-manager/src/lib/votecounting.ts

```diff
@@ -295,28 +276,24 @@ export function tallyVotesByContest({
       }
 
       // overvotes & undervotes
-      const maxSelectable =
-        contest.type === 'yesno' ? 1 : (contest as CandidateContest).seats
+      const maxSelectable = contest.type === 'yesno' ? 1 : contest.seats
       if (selected.length > maxSelectable || selected.length === 0) {
         return
       }
 
-      if (contest.type === 'yesno') {
-        const optionTally = find(tallies, (optionTally) => {
-          return (
-            (optionTally as YesNoContestOptionTally).option[0] === selected[0]
-          )
-        })
-        optionTally.tally += 1
-      } else {
-        ;(selected as CandidateVote).forEach((selectedOption) => {
-          const optionTally = find(tallies, (optionTally) => {
-            const candidateOption = optionTally.option as Candidate
-            const selectedCandidateOption = selectedOption as Candidate
-            return candidateOption.id === selectedCandidateOption.id
-          })
-          optionTally.tally += 1
-        })
+      for (const selectedOption of selected) {
+        tallies = tallies.map((optionTally) => ({
+          ...optionTally,
+          tally:
+            (isCandidateOption(optionTally.option) &&
+              isCandidateSelection(selectedOption) &&
+              optionTally.option.id === selectedOption.id) ||
```

Comment:
I think this could be faster if we cache'd and checked `isCandidateSelection(selectedOption)` and `isYesNoSelect(selectedOption)` first.

Overall I'm a tad worried about the (in)efficiency of looping through all the contest tallies to find the one to increment. That said, maybe this small fix is enough for now.

(I wonder how long this would take for 1M ballots, for example?)

Reply:
also wondering if we're doing a ton of object allocations for tallies that don't need to be modified.

Again, let's drive this from performance metrics on lots of ballots, and I'm okay if we improve performance in a followup PR, but I just want to flag this here.

Reply:
These changes were from brain's PR to make things more readonly, I intentionally was trying not to touch this code in this PR to reduce the surface area of change to minimize risk/etc. But I agree we should look at if there are ways to better optimize this in a followup PR. I'm going to remove the readonly markers just from ContestOptionTally and address making that readonly + optimizing this section of code as a followup item.